framework/src/org/checkerframework/common/jaifinference/JaifInferenceUtils.java

```diff
@@ -0,0 +1,485 @@
+            return BasicAFT.forType(String.class);
+        return null;
+    }
+
+    private static AnnotationMirror annotationToAnnotationMirror(
+            Annotation anno, ProcessingEnvironment processingEnv) {
+        final AnnotationBuilder builder = new AnnotationBuilder(processingEnv,
+                anno.def().name);
+        for (String fieldKey : anno.fieldValues.keySet()) {
+            addFieldToAnnotationBuilder(fieldKey,
+                    anno.fieldValues.get(fieldKey), builder);
+        }
+        return builder.build();
+    }
+
+    @SuppressWarnings("unchecked")
+    private static void addFieldToAnnotationBuilder(String fieldKey, Object obj,
+            AnnotationBuilder builder) {
+        if (obj instanceof List<?>) {
+            builder.setValue(fieldKey, (List<Object>) obj);
+        } else if (obj instanceof String) {
+            builder.setValue(fieldKey, (String)obj);
+        } else if (obj instanceof Integer) {
+            builder.setValue(fieldKey, (Integer)obj);
+        } else if (obj instanceof Float) {
+            builder.setValue(fieldKey, (Float)obj);
+        } else if (obj instanceof Long) {
+            builder.setValue(fieldKey, (Long)obj);
+        } else if (obj instanceof Boolean) {
+            builder.setValue(fieldKey, (Boolean)obj);
+        } else if (obj instanceof Character) {
+            builder.setValue(fieldKey, (Character)obj);
+        } else if (obj instanceof Class<?>) {
+            builder.setValue(fieldKey, (Class<?>)obj);
+        } else if (obj instanceof Double) {
+            builder.setValue(fieldKey, (Double)obj);
+        } else if (obj instanceof Float) {
+            builder.setValue(fieldKey, (Float)obj);
+        } else if (obj instanceof Enum<?>) {
+            builder.setValue(fieldKey, (Enum<?>)obj);
+        } else if (obj instanceof Enum<?>[]) {
+            builder.setValue(fieldKey, (Enum<?>[])obj);
+        } else if (obj instanceof AnnotationMirror) {
+            builder.setValue(fieldKey, (AnnotationMirror)obj);
+        } else if (obj instanceof Object[]) {
+            builder.setValue(fieldKey, (Object[])obj);
+        } else if (obj instanceof TypeMirror) {
+            builder.setValue(fieldKey, (TypeMirror)obj);
+        } else if (obj instanceof Short) {
+            builder.setValue(fieldKey, (Short)obj);
+        } else if (obj instanceof VariableElement) {
+            builder.setValue(fieldKey, (VariableElement)obj);
+        } else if (obj instanceof VariableElement[]) {
+            builder.setValue(fieldKey, (VariableElement[])obj);
+        } else {
+            throw new RuntimeException("Unrecognized type: " + obj.getClass());
+        }
+    }
+
+    /**
+     * Auxiliary method that returns the ClassSymbol of the class encapsulating
+     * the node n passed as parameter.
+     * TODO: This method could be moved somewhere else.
+     */
+    public static ClassSymbol getClassSymbol(ClassTree classTree,
```

Comment:
I would name this "getEnclosingClassSymbol".
Also, I don't feel that you need "auxiliary method that" in the Javadoc documentation.  Just say what it does.